source/common/network/listener_impl.h

```diff
@@ -47,6 +47,7 @@ class ListenerImpl : public Listener {
   const bool use_proxy_proto_;
   ProxyProtocol proxy_protocol_;
   const bool use_original_dst_;
+  const size_t per_connection_buffer_limit_bytes_;
```

Comment:
Just have a ListenerOptions as a member var and then you can just copy it in via passed in options.